libraries/fbx/src/GLTFSerializer.cpp

```diff
@@ -979,668 +1074,504 @@ bool GLTFSerializer::buildGeometry(HFMModel& hfmModel, const hifi::VariantHash&
-                            blendshape.normals.resize(prevMeshVerticesCount + vertices.size());
-                        }
-                        for (int i = 0; i < vertices.size(); i++) {
-                            blendshape.indices[prevMeshVerticesCount + i] = prevMeshVerticesCount + i;
-                            blendshape.vertices[prevMeshVerticesCount + i] += vertices.value(i);
-                            blendshape.normals[prevMeshVerticesCount + i] += normals.value(i);
-                        }
+                    QVector<glm::vec3> normals;
+                    QVector<glm::vec3> vertices;
 
+                    if (target.values.contains((QString) "NORMAL")) {
+                        generateTargetData(target.values.value((QString) "NORMAL"), weight, normals);
+                    }
+                    if (target.values.contains((QString) "POSITION")) {
+                        generateTargetData(target.values.value((QString) "POSITION"), weight, vertices);
+                    }
+                    if (blendshape.indices.size() < prevMeshVerticesCount + vertices.size()) {
+                        auto targetSize = prevMeshVerticesCount + vertices.size();
+                        blendshape.indices.resize(targetSize);
+                        blendshape.vertices.resize(targetSize);
+                        blendshape.normals.resize(targetSize);
+                    }
+                    for (int i = 0; i < vertices.size(); i++) {
+                        blendshape.indices[prevMeshVerticesCount + i] = prevMeshVerticesCount + i;
+                        blendshape.vertices[prevMeshVerticesCount + i] += vertices.value(i);
+                        blendshape.normals[prevMeshVerticesCount + i] += normals.value(i);
                     }
                 }
+            }         
+        }
+    }
 
-                foreach(const glm::vec3& vertex, mesh.vertices) {
-                    glm::vec3 transformedVertex = glm::vec3(globalTransforms[nodeIndex] * glm::vec4(vertex, 1.0f));
-                    mesh.meshExtents.addPoint(transformedVertex);
-                    hfmModel.meshExtents.addPoint(transformedVertex);
-                }               
-            }
+    // Create the instance shapes for each transform node
+    for (int nodeIndex = 0; nodeIndex < numNodes; ++nodeIndex) {
+        const auto& node = _file.nodes[nodeIndex];
+        if (-1 == node.mesh) {
+            continue;
+        }
 
-            // Mesh extents must be at least a minimum size, in particular for blendshapes to work on planar meshes.
-            const float MODEL_MIN_DIMENSION = 0.001f;
-            auto delta = glm::max(glm::vec3(MODEL_MIN_DIMENSION) - mesh.meshExtents.size(), glm::vec3(0.0f)) / 2.0f;
-            mesh.meshExtents.minimum -= delta;
-            mesh.meshExtents.maximum += delta;
-            hfmModel.meshExtents.minimum -= delta;
-            hfmModel.meshExtents.maximum += delta;
```

Comment:
this sizing code was changed here: https://github.com/kasenvr/project-athena/pull/671

but removed in this original PR?  I think it has moved to thickenFlatExtents in HFMModelMath but now maybe it's lost david's changes?